crates/api/src/site/search.rs

```diff
@@ -64,7 +70,9 @@ impl Perform for Search {
     } else {
       None
     };
-    let creator_id = data.creator_id;
+
+    check_page_and_limit(page, limit)?;
```

Comment:
Actually i think it would be better to move these checks into PostQueryBuilder etc, and same for check_missing_community_id(). Otherwise its easy to forget when adding a new query somewhere.